content/techniques/file-upload.md

```diff
@@ -41,6 +41,113 @@ The `FileInterceptor()` decorator takes two arguments:
 
+#### File Validation
+
+Often times it can be useful to validate incoming file metadata, like file size or file mimetype. You can create your own [`Pipes`](https://docs.nestjs.com/pipes) and add them to the `FileInterceptor` decorator to achieve this, considering the `value` inside each pipe will be the file object. The example below demonstrates how a basic file size validator pipe could be implemented:
+
+```typescript
+
+import { PipeTransform, Injectable, ArgumentMetadata } from '@nestjs/common';
+
+@Injectable()
+export class ValidationPipe implements PipeTransform {
+  transform(value: any, metadata: ArgumentMetadata) {
+    // "value" is actually the object containing the file's attributes and metadata.
+    const oneKb = 1000
+    return value.size < oneKb
+  }
+}
+
+```
+
+Nest provides a built-in pipe to handle common use cases and facilitate / standardize the addition of new ones. This pipe is called `ParseFilePipe`, and looks like the following:
+
+```typescript
+  @Post('file')
+  uploadFileAndPassValidation(
+    @Body() body: SampleDto,
+    @UploadedFile(
+      new ParseFilePipe({
+        validators: [
+          // ... Set of file validator instances here
+        ]
+      })
+    )
+    file: Express.Multer.File,
+  ) {
+    return {
+      body,
+      file: file.buffer.toString(),
+    };
+  }
+```
+
+As you can see it's necessary to specify an array of file validators as one of the `ParseFilePipe`'s options. We'll discuss the interface of a validator, but it's worth mentioning this pipe also has two additional **optional** options:
+
+| `errorHttpStatusCode` | The HTTP status code to be thrown in case **any** validator fails. Default is `400` (BAD REQUEST). |
+|---------------------------------|----------------------------------------------------------------------------------------------------|
+| `exceptionFactory`    | A factory which receives the error message and returns an error.                                   |
+
+Now, back to the `FileValidator` interface. To use validators in this pipe you have to either use built-in implementations or provide your own custom `FileValidator`. The code below describes such interface and its explanation:
+
+```typescript
+export abstract class FileValidator<TValidationOptions = Record<string, any>> {
+  constructor(protected readonly validationOptions: TValidationOptions) {}
+
+  /**
+   * Indicates if this file should be considered valid, according to the options passed in the constructor.
+   * @param file the file from the request object
+   */
+  abstract isValid(file?: any): boolean | Promise<boolean>;
+
+  /**
+   * Builds an error message in case the validation fails.
+   * @param file the file from the request object
+   */
+  abstract buildErrorMessage(file: any): string;
+}
+```
+
+> info **Hint** The `FileValidator` interfaces supports async validation via its `isValid` function.
```

Comment:
```suggestion
> info **Hint** The `FileValidator` interfaces supports async validation via its `isValid` function. To leverage type security, you can also type the `file` parameter as `Express.Multer.File` in case you are using express (default) as a driver.
```